working/unquoted-imports/feature-specification.md

```diff
@@ -281,7 +315,37 @@ import strange /* comment */    .   but
       fine;
 ```
 
-But they can also choose to *not* do that.
+This wouldn't cause any problems for a Dart implementation. It would simply
+discard the whitespace and comments as it does elsewhere and the resulting path
+is `strange.but/another/fine`.
+
+However, it likely causes problems for Dart *users* and other simpler tools and
+scripts that work with Dart code. In particular, we often see homegrown tools
+that want to "parse" a Dart file to find its package references and traverse the
+dependency graph. While these tools ideally should use a full Dart parser (like
+the one in the [analyzer package][], which is freely available), the reality is
+that users often cobble together simple scripts using regex to do this kind of
+parsing, or they need to write these tools in a language other than Dart. In
+those cases, if the package path happens to contain whitespace or comments, the
+tool will likely silently fail to recognize the package path.
+
+[analyzer package]: https://pub.dev/packages/analyzer
+
+Also, we find no compelling *use* for whitespace and comments inside package
+paths. To that end, this proposal makes it an error. All of the tokens in the
```

Comment:
Good point.

It's somewhat like inverted pyramid writing. Start by saying what this document is (it's the design document for a Dart language feature that will allow shorter syntax for common import URIs), then the conclusion and consequences. Then the actual specification. Then support it with further arguments and discussions. Then support it with extra data, graphs, etc., maybe in appendices. The goal is to give readers clear places to *get off* the reading train, when they have all they need, and so that they don't miss anything essential by doing so. If you don't care about specification for import URIs, you can stop after the summary. If you just want to know the big lines, you can stop after the first step. If you just want the spec, you can stop after step 2. If you want the *why*, you can keep reading.

(Academic writing tend to spend a lot of time on set-up and motivation. Design documents are about *decisions* first. The things we didn't choose are not going to be important to any of our users in the long run. It's nice to have the discussion of *why* things were chosen, so a later generation won't think "there was probably a good reason for this completely arbitrary decision" when it really was "it's as good as any, and we need one". But most readers just want the conclusion.)

Reply:
I have many feelings about inverted pyramid and narrative order. :)

It totally makes sense to put the conclusions up front and let people get off the train early when they have the information they need.

At the same time, my experience is that in persuasive writing, it can be more effective to delay the conclusion until later in the narrative. When you state it up front, if the reader has a knee-jerk negative reaction to it, they will resist the rest of the prose and poke holes at it. If you lead them gently to the conclusion along a path of things they agree with, when they reach the end, it often sinks in more effectively.

The challenge is that language proposals are doing a mixture of informing and persuading, so figuring out the right order is hard.

But in this case, I think Kallen has a good point. Added a couple of sentences up front. I didn't add anything to the reserved words section since the previous section already states that reserved words and built-in identifiers are allowed.